internal/pkg/metadb/metadb.go

```diff
@@ -831,6 +830,9 @@ func (m *MetaDB) QueryRecords(ctx context.Context, req *pb.QueryRecordsRequest)
 	if limit := req.GetLimit(); limit > 0 {
 		query = query.Limit(int(limit))
 	}
+	if offset := req.GetOffset(); offset > 0 {
+		query = query.Offset(int(offset))
```

Comment:
Can we enforce this by placing the offset check after `req.GetKeysOnly`?

Reply:
I'd like to keep this implementation as is (temporarily) to enable our current client to pull records within a store that has thousands of entries.
Pros: the client is not timing out waiting for the results or downloading them locally (think about pulling 100 records at index 22000 in a store, which is a scenario we see).
Cons: the client ends up generating get requests for all these records.

I am planning to address this after I introduce a GetMulti endpoint in metadb. With it, I'll be able to enforce that the keys oly are pulled with offset>0, and the GetMulti is used to actually get the record content. The reasoning is to keep this entirely in Open Saves without forcing the client to pull the keys and then call again to get the records it wants.

Reply:
The only issue I have is that removing the ability to query by offset _without_ keys only is a breaking change, even if we explicitly disallow this in the comment. At the end of the day, behavior is different. Could you use this change in a branch until the new endpoint is added?

If not, the only other way I could potentially see this is if we explicitly mark the `Offset` field as experimental and subject to change (don't depend on it). The comment could be like 
```// It is EXPERIMENTAL and subject to change or removal without notice.```

Reply:
Yes, I see your point about breaking change.

I'll see about introducing a GetMulti function to take care of this optimization in this PR, and keep the behavior as is. I.e. when the client asks for keysOnly, they get the key. When they don't, we still use the offset to query keys internally but also calls a new GetMulti function on the results to returns full records. Thus keeping backward compatibility but also enabling querying full records with an offset (even though if we query by keysOnly internally).